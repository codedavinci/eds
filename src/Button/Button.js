import React from 'react'
import PropTypes from 'prop-types'

const Button = ({ bg, children, ...rest }) => (
  <button style={{ backgroundColor: bg }} {...rest}>
    {children}
  </button>
)

Button.propTypes = {
  bg: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
}

Button.displayName = 'Button'

export default Button
