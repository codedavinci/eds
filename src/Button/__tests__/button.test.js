import React from 'react'
import Button from '../Button'

describe('<Button />', () => {
  const defaultProps = {
    bg: 'pink',
    children: 'Hello',
  }

  it('should render', () => {
    const btn = shallow(<Button {...defaultProps} />)
    expect(btn).toMatchSnapshot()
  })
})
