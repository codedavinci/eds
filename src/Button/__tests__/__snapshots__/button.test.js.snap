// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render 1`] = `
<button
  style={
    Object {
      "backgroundColor": "pink",
    }
  }
>
  Hello
</button>
`;
